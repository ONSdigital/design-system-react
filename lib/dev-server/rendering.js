import fs from "fs";

const DEFAULT_PARAMS = {
  bodyClass: "",
};

export function renderPage(params) {
  const templatePath = "./lib/dev-server/template.html";
  const template = fs.readFileSync(templatePath, { encoding: "utf8" });

  params = Object.assign({}, DEFAULT_PARAMS, params);

  let result = template;
  for (let paramName of Object.keys(params)) {
    result = result.replace(new RegExp(`[{][{]${paramName}[}][}]`, "g"), params[paramName]);
  }

  return result;
}
