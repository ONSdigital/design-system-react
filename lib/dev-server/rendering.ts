import fs from "fs-extra";

const DEFAULT_PARAMS = {
  designSystemVersion: "1.2.3",
  bodyClass: "",
};

export function renderPage(params: any): string {
  const templatePath = "./lib/dev-server/template.html";
  const template = fs.readFileSync(templatePath, { encoding: "utf8" });

  params = Object.assign({}, DEFAULT_PARAMS, params);

  let result = template;
  for (let paramName of Object.keys(params)) {
    result = result.replace(new RegExp(`[{][{]${paramName}[}][}]`, "g"), params[paramName]);
  }

  return result;
}
