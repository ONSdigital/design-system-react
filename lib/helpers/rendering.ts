import fs from "fs-extra";

const DEFAULT_PARAMS = {
  designSystemVersion: "1.2.3",
};

export function loadTemplate(templatePath: string): string {
  return fs.readFileSync(templatePath, { encoding: "utf8" });
}

export function renderPage(template: string, params: any): string {
  params = Object.assign({}, DEFAULT_PARAMS, params);

  let result = template;
  for (let paramName of Object.keys(params)) {
    result = result.replace(new RegExp(`[{][{]${paramName}[}][}]`, "g"), params[paramName]);
  }

  return result;
}
