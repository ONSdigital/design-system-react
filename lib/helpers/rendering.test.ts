import { loadTemplate, renderPage } from "./rendering";

const FAKE_TEMPLATE = `
  <!doctype html>
  <html lang="en">
    <head>
      <meta charset="utf-8">
      <title>{{title}} - Design System React</title>
      <link rel="stylesheet" href="https://cdn.ons.gov.uk/sdc/design-system/61.0.4/css/main.css">
      <link rel="stylesheet" href="/main.css">
    </head>
    <body class="{{bodyClass}}">
      {{body}}
    </body>
  </html>
`;

describe("dev-server/rendering", () => {
  describe("loadTemplate(templatePath: string): string", () => {
    it("loads the template", () => {
      const template = loadTemplate("./lib/helpers/example-template.html");

      expect(template).toMatchSnapshot();
    });
  })

  describe("renderPage(template: string, params: any): string", () => {
    it("substitutes placeholder values", () => {
      const html = renderPage(FAKE_TEMPLATE, {
        designSystemVersion: "1.2.3",
        title: "An example page",
        bodyClass: "an-example-body-class",
        body: `
          <p>Example body content.</p>
        `,
      });

      expect(html).toMatchSnapshot();
    });
  });
});
