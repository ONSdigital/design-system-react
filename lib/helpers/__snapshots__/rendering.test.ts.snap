// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dev-server/rendering loadTemplate(templatePath: string): string loads the template 1`] = `
"<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <title>{{title}} - Example Template</title>
    <link rel="stylesheet" href="/main.css">
  </head>
  <body>
    {{body}}
  </body>
</html>
"
`;

exports[`dev-server/rendering renderPage(template: string, params: any): string substitutes placeholder values 1`] = `
"
  <!doctype html>
  <html lang="en">
    <head>
      <meta charset="utf-8">
      <title>An example page - Design System React</title>
      <link rel="stylesheet" href="https://cdn.ons.gov.uk/sdc/design-system/61.0.4/css/main.css">
      <link rel="stylesheet" href="/main.css">
    </head>
    <body class="an-example-body-class">
      
          <p>Example body content.</p>
        
    </body>
  </html>
"
`;
